colorizer_data/writer.py

```diff
@@ -483,6 +484,29 @@ def validate_dataset(
 
         # TODO: Add validation for other required data files
 
+        # Check that all features are unique.
+        # TODO: Use dictionary to store features during writing?
```

Comment:
This todo sounds like a good approach to me!

Reply:
Ended up just implementing this so it doesn't have to wait for another PR :)